### PR TITLE
HttpRetryPolicy: Fixes Request Timeout configuration in HttpTimeoutPolicyNoRetry

### PR DESCRIPTION
**Pull request by sourabh1007**
## Description

Right Now, NoRetry policy does not have request timeout value which is leading to request cancellation. Fixing this issue as part this PR.

## Type of change
- [] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/HttpClient/CosmosHttpClientCore.cs [hunk 1]
```diff
@@ -265,7 +265,7 @@ ValueTask<HttpRequestMessage> CreateRequestMessage()
                             return responseMessage;
                         }
 
-                        bool isOutOfRetries = CosmosHttpClientCore.IsOutOfRetries(timeoutPolicy, startDateTimeUtc, timeoutEnumerator);
+                        bool isOutOfRetries = timeoutPolicy is HttpTimeoutPolicyNoRetry || CosmosHttpClientCore.IsOutOfRetries(timeoutPolicy, startDateTimeUtc, timeoutEnumerator);
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/HttpClient/CosmosHttpClientCore.cs:268]: Why is this needed? It should just return false.

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/HttpClient/CosmosHttpClientCore.cs:268]: In case of `NoRetryPolicy `, why to even call any function to check outofretries

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/HttpClient/CosmosHttpClientCore.cs:268]: It violates the ownership of the logic. Now anywhere else that would use these classes would have to implement the same custom logic. If it is handled like all the other retry policies then no custom logic is needed. This is only called in the exception scenario so perf should not be an issue with calling a method.

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/HttpClient/CosmosHttpClientCore.cs:268]: Retry policy is meant for only failure scenario. This condition is here just to behave like `NoOp` as retries concept is not applicable for NoRetry policies.
If you still have concern with this, I can remove this condition. I don't think this code change is even worth discussing.

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/HttpClient/CosmosHttpClientCore.cs:268]: I don't understand why this needs to exist. The timeout policy should not have any retries so this should return false. There shouldn't need to be any custom handling for this policy.

**Review by j82w — APPROVED** [review]
